Fixes CLI in Azure Cloud Shell. Closes: #6117

Adam-it opened the pull request:
## 🔗 Related

Closes: #6117 

## 🧪 How to test

Since testing CLI from local build is kinda tricky in Azure Cloud Shell I will add some more details here with instructions and some result

1. got to portal.azure and run the Azure Cloud Shell
2. git clone my repo running `git clone https://github.com/Adam-it/cli-microsoft365.git` and checkout the branch with this change `git checkout bug-fix/azure-cloud-shell-issue` (this may be done differently of course) 
3. run insall and build CLI `npm install; npm run clean; npm run build;`. Running `npm link` won't work as we do not have admin permissions in Azure Cloud Shell
4. go do dist folder `cd ./dist/ `
5. test CLI commands running `node index.js` instead of `m365`. For example to try out `m365 status` command run `node index.js status`

## 🤔 What should be tested

Since we remove the spinner in general everything 😜. But for sure we need to recheck `chili` and `external connection doctor` command.

## 📷 Result

![image](https://github.com/pnp/cli-microsoft365/assets/58668583/00397853-0bab-4c34-84ef-9a8c16d2f7a0)

waldekmastykarz reviewed (changes requested):
We need to do some changes. Right now, the `external connection doctor` command, doesn't produce any output in text mode:

```text
$ m365 external connection doctor -i foodstore -o text

```

The same in json output:

```text
$ m365 external connection doctor -i foodstore -o json
[
  {
    "id": "loadExternalConnection",
    "text": "Load connection",
    "type": "required",
    "status": "passed"
  },
  {
    "id": "loadSchema",
    "text": "Load schema",
    "type": "required",
    "status": "passed"
  },
  {
    "id": "copilotRequiredSemanticLabels",
    "text": "Required semantic labels",
    "type": "required",
    "errorMessage": "Missing label iconUrl",
    "status": "failed"
  },
  {
    "id": "searchableProperties",
    "text": "Searchable properties",
    "type": "required",
    "status": "passed"
  },
  {
    "id": "contentIngested",
    "text": "Items have content ingested",
    "type": "required",
    "status": "passed"
  },
  {
    "id": "enabledForInlineResults",
    "text": "Connection configured for inline results",
    "type": "required",
    "status": "manual"
  },
  {
    "id": "itemsHaveActivities",
    "text": "Items have activities recorded",
    "type": "recommended",
    "status": "manual"
  },
  {
    "id": "meaningfulNameAndDescription",
    "text": "Meaningful connection name and description",
    "type": "required",
    "status": "manual"
  },
  {
    "id": "semanticLabels",
    "text": "Semantic labels",
    "type": "recommended",
    "status": "passed"
  },
  {
    "id": "resultType",
    "text": "Result type",
    "type": "recommended",
    "status": "passed"
  },
  {
    "id": "urlToItemResolver",
    "text": "urlToItemResolver configured",
    "type": "recommended",
    "status": "passed"
  }
]
```

martinlingstuyl commented on src/Auth.ts:
```diff
@@ -456,16 +456,7 @@ export class Auth {
-    if (!cli.spinner.isSpinning && typeof global.it === 'undefined') {
-      cli.spinner.start();
-    }
+    await logger.log(response.message);
```
As this is not command output, we should move it to `stdErr`. Except when output is configured to go to `stdout`... Just like the inquirer prompt (check out `prompt.ts`)

waldekmastykarz replied:
Good catch @martinlingstuyl 👏

Adam-it replied:
Thanks for comment. It was like this before we moved it to spinner that's why I reverted it to this state. But I will change 👍

Adam-it commented:
> We need to do some changes. Right now, the `external connection doctor` command, doesn't produce any output in text mode:
> 
> ```
> $ m365 external connection doctor -i foodstore -o text
> ```
> 
> The same in json output:
> 
> ```
> $ m365 external connection doctor -i foodstore -o json
> [
>   {
>     "id": "loadExternalConnection",
>     "text": "Load connection",
>     "type": "required",
>     "status": "passed"
>   },
>   {
>     "id": "loadSchema",
>     "text": "Load schema",
>     "type": "required",
>     "status": "passed"
>   },
>   {
>     "id": "copilotRequiredSemanticLabels",
>     "text": "Required semantic labels",
>     "type": "required",
>     "errorMessage": "Missing label iconUrl",
>     "status": "failed"
>   },
>   {
>     "id": "searchableProperties",
>     "text": "Searchable properties",
>     "type": "required",
>     "status": "passed"
>   },
>   {
>     "id": "contentIngested",
>     "text": "Items have content ingested",
>     "type": "required",
>     "status": "passed"
>   },
>   {
>     "id": "enabledForInlineResults",
>     "text": "Connection configured for inline results",
>     "type": "required",
>     "status": "manual"
>   },
>   {
>     "id": "itemsHaveActivities",
>     "text": "Items have activities recorded",
>     "type": "recommended",
>     "status": "manual"
>   },
>   {
>     "id": "meaningfulNameAndDescription",
>     "text": "Meaningful connection name and description",
>     "type": "required",
>     "status": "manual"
>   },
>   {
>     "id": "semanticLabels",
>     "text": "Semantic labels",
>     "type": "recommended",
>     "status": "passed"
>   },
>   {
>     "id": "resultType",
>     "text": "Result type",
>     "type": "recommended",
>     "status": "passed"
>   },
>   {
>     "id": "urlToItemResolver",
>     "text": "urlToItemResolver configured",
>     "type": "recommended",
>     "status": "passed"
>   }
> ]
> ```

I worked on this a bit. Initially, I didn't notice we used spinner for `text` response of this command. I refactored it to same approach what we use in `spfx doctor` command and added `information` and `warning` response types to support the different options we have for this command 

this is how  it works both for `text` and `json` output mode

![Screenshot 2024-07-06 235236](https://github.com/pnp/cli-microsoft365/assets/58668583/cbaf7fca-1a9a-4e42-9189-1282332fe665)

![Screenshot 2024-07-06 235329](https://github.com/pnp/cli-microsoft365/assets/58668583/fed01fe2-80f8-41dd-911e-6c6f34426895)

Jwaegebaert commented on src/Auth.ts:
```diff
@@ -456,16 +456,7 @@ export class Auth {
-    if (!cli.spinner.isSpinning && typeof global.it === 'undefined') {
-      cli.spinner.start();
-    }
+    await logger.logToStderr(`🌶️  ${response.message}`);
```
Questioning a bit why we always have two spaces between the🌶️ and our message. I noticed this was from the start already but never quite understood why 😄

Jwaegebaert reviewed (approved):
Sad to see the spinner go, but it's causing too many issues atm. Nice work with this @Adam-it, and great explanation on how to pull it onto the Azure Cloud Shell! I left one remark, but it's more of a question than something that needs changing.

milanholemans commented:
🪦🌹 Spinner 😢
